engine/src/main/java/org/terasology/math/Diamond3iIterable.java

```diff
@@ -102,26 +102,26 @@ public boolean hasNext() {
 
             @Override
             public Vector3ic next() {
-                pos.set(origin.x + offset[0], origin.y + offset[1], origin.z + offset[2]);
-                if (offset[2] < 0) {
-                    offset[2] *= -1;
-                } else if (offset[1] < 0) {
-                    offset[1] *= -1;
-                    offset[2] = -(level[0] - TeraMath.fastAbs(offset[0]) - TeraMath.fastAbs(offset[1]));
+                pos.set(origin.x + offset.x, origin.y + offset.y, origin.z + offset.z);
```

Comment:
As suggested before, why not use vector math?
```suggestion
                origin.add(offset, pos);
```
alternatively: `pos.set(origin).add(offset)`